### PR TITLE
Feature-411 Parathreads list screen

### PR DESCRIPTION
**Pull request by ishrak-imam**
### Description
Parathreads list screen

### CODE REVIEW

File: src/screen/Drawer/DrawerScreen.tsx [hunk 1]
```diff
@@ -29,6 +30,12 @@ function DrawerScreen({navigation}: DrawerContentComponentProps) {
           </TouchableOpacity>
         </View>
         <Layout style={styles.rest}>
+          <ListItem
+            title="Accounts"
+            accessoryLeft={(props) => <Icon {...props} name="person-outline" animation="zoom" />}
+            onPress={() => navigation.navigate(accountsNavigator)}
+          />
+          <Divider />
```
Comment (ishrak-imam) [inline src/screen/Drawer/DrawerScreen.tsx]: @ahoseinian I am not sure how this change got in here.
Can you please confirm if it is ok?

Comment (silva-fj) [reply src/screen/Drawer/DrawerScreen.tsx]: Since the dashboard is our main entry it makes sense to be first in the Drawer screen. You can move `Accounts` back

Comment (ahoseinian) [reply src/screen/Drawer/DrawerScreen.tsx]: hmm probably some problem in merge, but maybe you can move it under dashboard. I thought dashboard being the top must item is better.

Comment (ishrak-imam) [reply src/screen/Drawer/DrawerScreen.tsx]: `Accounts` menu item is after dashboard now.
https://github.com/litentry/litentry-app/pull/443/commits/53671380409ec16635731e459ca0337273734f46

File: src/api/hooks/useParathreadInfo.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import type {Codec} from '@polkadot/types/types';
+
+export function sliceHex(value: Codec, max = 6): string {
+  const hex = value.toHex();
+  return hex.length > 2 * max + 2 ? `${hex.slice(0, max + 2)}…${hex.slice(-max)}` : hex === '0x' ? '' : hex;
+}
+
+export function useParathreadInfo(id: ParaId) {
+  return useApiQuery(['parathreadInfo', id], async (api: ApiPromise) => {
+    const paraInfo = await api.query.registrar?.paras?.<Option<ParaInfo>>(id);
```
Comment (ahoseinian) [inline src/api/hooks/useParathreadInfo.ts]: maybe do a promise.all here?

File: src/screen/ParathreadsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  name?: string;
+  homepage?: string;
+};
+
+export function ParathreadsScreen() {
+  const {data: leasePeriod} = useParachainsLeasePeriod();
+  const {data: parathreads, isLoading} = useParathreads();
+
+  const toHomepage = (url?: string) => {
+    if (url) {
```
Comment (ahoseinian) [inline src/screen/ParathreadsScreen.tsx]:
```suggestion
    if (url && Linking.canopenurl(url)) {
```

File: src/screen/ParathreadsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    if (url && Linking.canOpenURL(url)) {
+      Linking.openURL(url);
+    }
+  };
+
+  return (
+    <SafeView edges={noTopEdges}>
+      {isLoading ? (
+        <View style={globalStyles.centeredContainer}>
+          <Spinner />
```
Comment (silva-fj) [inline src/screen/ParathreadsScreen.tsx]: We have a `LoadingView` component, let's use that one

File: src/screen/ParathreadsScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+type ParathreadData = {
+  name?: string;
+  homepage?: string;
+};
+
+export function ParathreadsScreen() {
+  const {data: leasePeriod} = useParachainsLeasePeriod();
+  const {data: parathreads, isLoading} = useParathreads();
+
+  const toHomepage = (url?: string) => {
```
Comment (silva-fj) [inline src/screen/ParathreadsScreen.tsx]: Let's move this function out of this component and use it directly in `ParathreadItem` instead of passing it as a prop to all items

File: src/screen/ParathreadsScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+          <Spinner />
+        </View>
+      ) : (
+        leasePeriod && (
+          <FlatList
+            style={globalStyles.flex}
+            contentContainerStyle={styles.content}
+            keyExtractor={([id, _]) => id.toString()}
+            data={parathreads}
+            renderItem={({item}) => <ParathreadItem id={item[0]} leases={item[1]} toHomepage={toHomepage} />}
```
Comment (silva-fj) [inline src/screen/ParathreadsScreen.tsx]: NIT: It would be better if we get objects instead of arrays in `parathreads`, so we ca do `item.id` and `item.leases`

File: src/screen/ParathreadsScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+                      {identity.display}
+                    </Text>
+                  </View>
+                )}
+              </View>
+            )}
+          </Account>
+        </View>
+      )}
+      {/* @TODO: Add leases information */}
```
Comment (silva-fj) [inline src/screen/ParathreadsScreen.tsx:107]: Are you planning to do this in another PR?

Comment (ishrak-imam) [reply src/screen/ParathreadsScreen.tsx:107]: Right now there are no lease information for the parathreads so I don't have an example of how the data is represented in the UI. If they are the same as parachains I guess it can be done but I decided to skip it for now until we have some actual lease data and final UI design.

So yes, maybe later with another PR, maybe I can create a ticket for it? 🤔

Comment (silva-fj) [reply src/screen/ParathreadsScreen.tsx:107]: Yeah let's create an issue so we don't forget 👍🏼

**Review by ahoseinian — APPROVED** [review]

**Review by silva-fj — APPROVED** [review]
